### PR TITLE
[Experimental] Upgrade To Rails 5.2

### PR DESCRIPTION
**Pull request by dinesh-skyach**
Upgrade To Rails 5.2

### CODE REVIEW

**Comment by ianeinser** [comment]
Bring it on dudes ASAP, for both v1.9 and v2.0 please.

**Comment by shal** [comment]
@dinesh-skyach Could you rebase the patch?

File: Gemfile [hunk 1]
```diff
@@ -2,83 +2,87 @@
...
+  gem 'bullet', '~> 5.9.0'
   gem 'grape_on_rails_routes', '~> 0.3.2'
 end
 
 group :development do
-  gem 'annotate',   '~> 2.7'
-  gem 'ruby-prof',  '~> 0.17.0', require: false
+  gem 'annotate', '~> 2.7.4'
+  gem 'ruby-prof', '~> 0.17.0', require: false
+  gem 'spring', '~> 2.0.2'
```
Comment (shal) [inline Gemfile:72]: @dinesh-skyach Why do we need this gems: `spring` and `spring-watcher-listen`?

File: Gemfile [hunk 2]
```diff
@@ -2,83 +2,87 @@
...
-  gem 'rubocop-rspec',       '~> 1.30', require: false
+  gem 'rspec-rails', '~> 3.8.1'
+  gem 'rspec-retry', '~> 0.6.1'
+  gem 'webmock', '~> 3.5.1'
+  gem 'database_cleaner', '~> 1.7.0'
+  gem 'mocha', '~> 1.7.0', require: false
+  gem 'factory_bot_rails', '~> 4.11.1'
+  gem 'timecop', '~> 0.9.1'
+  gem 'rubocop-rspec', '~> 1.31.0', require: false
+  gem 'rails-controller-testing', '~> 1.0.4'
```
Comment (shal) [inline Gemfile:85]: I think later we should have a separate task to get rid of `rails-controller-testing`.
https://github.com/rails/rails/issues/1895

File: app/models/order.rb [hunk 3]
```diff
@@ -35,7 +35,7 @@ class Order < ActiveRecord::Base
 
   after_commit on: :update do
     next unless ord_type == 'limit'
-    event = case previous_changes.dig('state', 1)
+    event = case state
```
Comment (shal) [inline app/models/order.rb:38]: Maybe let's keep:

```ruby
event = case after_save.dig('state', 1)
```

File: bin/setup [hunk 4]
```diff
@@ -3,37 +3,46 @@
 # encoding: UTF-8
 # frozen_string_literal: true
 
-# path to your application root
-APP_ROOT = File.expand_path('../../', __FILE__)
+require 'fileutils'
+include FileUtils
```
Comment (shal) [inline bin/setup:7]: I don't this is it necessary to include and require `FileUtils`. Just keep 
```rb
Dir.chdir APP_ROOT do
  ...
end
```

File: config/application.rb [hunk 5]
```diff
@@ -1,11 +1,9 @@
 # encoding: UTF-8
 # frozen_string_literal: true
 
-require File.expand_path('../boot', __FILE__)
+require_relative 'boot'
 
-require 'rails'
-
-%w( active_record action_controller action_view sprockets ).each { |framework| require "#{framework}/railtie" }
+require 'rails/all'
```
Comment (shal) [inline config/application.rb:6]: I afraid amount of unused libs may speed down the application.

Comment (shal) [reply config/application.rb:6]: Keep this:

```rb
require_relative 'boot'

require 'rails'

%w( active_record action_controller action_view sprockets ).each { |framework| require "#{framework}/railtie" }
```

File: config/credentials.yml.enc [hunk 6]
```diff
@@ -0,0 +1 @@
+nvyOF/IyR5uU8ouBSuMUepwoSYJbiAJ5uK+jXqp/Uy2s0BEktlGaRACEBxu+rAdiSNVAwKfm+/2ZSiBd4QXTZa8YGLOPTBjKhL9xoT8EFtLLngTp2KIz367L4dtCWZ2Cdv9Fqf4bnZtSKnPXk4oECN0OnsCmiUInZW4bVU9E5jFqTsAyzuD8P1KOkmFvjJAHxbuO/w1tuLiKW253DZhdtbW4LHABgUFagnPUym6jHnMec272zfjsrFhYSjFWeWGg1qIBwRow/l3f30HR93X0cXqCWv5CCgDOAElrxQEWJ3P5SnYEydt9t8IDvw9cSnmS5aPpzMxOqw7PB7bA4I8BW2tgmpi364I1UWSU1Nd2co3hOR5pmIpb7etbJJfC1jw/4Ai9qVQHeMqNXUDBO2uVR9JbKq7v3nnjP1A1--d4xxvUu/a12DL/sR--sJ1yQRmZFkqaGVQYVM1J7A==
```
Comment (shal) [inline config/credentials.yml.enc:1]: Why do we need this?

File: config/environments/development.rb [hunk 7]
```diff
@@ -14,23 +14,34 @@
   # Do not eager load code on boot.
   config.eager_load = false
 
-  # Show full error reports and disable caching.
-  config.consider_all_requests_local       = true
+  # Show full error reports.
+  config.consider_all_requests_local = true
```
Comment (shal) [inline config/environments/development.rb:18]: No need.

File: config/environments/development.rb [hunk 8]
```diff
@@ -14,23 +14,34 @@
...
   # number of complex assets.
   config.assets.debug = false
 
   config.active_record.default_timezone = :local
 
-  # Asset digests allow you to set far-future HTTP expiration dates on all assets,
-  # yet still be able to expire them through the digest params.
-  config.assets.digest = true
+  # Suppress logger output for asset requests.
+  config.assets.quiet = true
```
Comment (shal) [inline config/environments/development.rb:44]: Try:
```rb
config.logger = ActiveSupport::Logger.new(STDOUT)
config.assets.quiet = true
```

**Review by shal — CHANGES_REQUESTED** [review]

**Comment by shal** [comment]
@dinesh-skyach @mod In my opinion this patch is very dangerous, it is too difficult to review all this changes and I think impossible to check everything.
I suggest to move slowly the version of rails.

Here is quote from rails main page:
https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#the-upgrade-process.

Let start from https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-4-2-to-rails-5-0 and then keep bumping the version slowly :smile_cat:

**Comment by shal** [comment]
Closed via #2095.

**Pull request closed without merging**